examples/docs_snippets/docs_snippets/concepts/resources/resources.py

```diff
@@ -203,3 +203,38 @@ def use_db_connection(context):
 
 
 # end_cm_resource_op
+
+
+@job
+def the_job():
+    ...
+
+
+def get_the_db_connection(creds):
+    ...
+
+
+# start_build_resources_example
+from dagster import resource, build_resources, sensor
+
+
+@resource
+def the_credentials():
+    ...
+
+
+@resource(required_resource_keys={"credentials"})
+def the_db_connection(init_context):
+    get_the_db_connection(init_context.resources.credentials)
+
+
+@sensor(job=the_job)
+def uses_db_connection():
+    with build_resources(
+        {"db_connection": the_db_connection, "credentials": credentials}
+    ) as resources:
+        conn = resources.db_connection
+        ...
+
```

Comment:
i think it'd be super helpful to include this example on the sensor page! 
i.e. moving this example to https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#examples

my thinking was if i wanna use a resource in a sensor, the first place i'd look would be the sensor page, e.g. trying to find something like "required_resource_keys" on sensors, and then this example can help directly, if i wonder why, i can then learn about build_resource outside execution in [1].

as for the code snippet in [1], i guess 
```
with build_resources(
        {"db_connection": the_db_connection, "credentials": credentials}
    ) as resources:
        conn = resources.db_connection
        ...
```
could be sufficient to show the usage. or just duplicating this example in two places is fine imo.